app/models/post.rb

```diff
@@ -2,4 +2,25 @@ class Post < ApplicationRecord
   has_many :comments, foreign_key: post_id
   has_many :likes, foreign_key: post_id
   belongs_to :author, class_name: 'User', foreign_key: author_id
+
+  after_initialize do |_post|
+    update_user_post_counter unless Post.exists?(id:)
+  end
+
+  def increment_likes_counter
+    update(likes_counter: self.likes_counter += 1)
+  end
+
+  def increment_comments_counter
+    update(comments_counter: self.comments_counter += 1)
+  end
+
+  def update_user_post_counter
+    author.increment_posts_counter
+  end
+
+  def most_recent_comments(limit = 5)
+    p 'Hello'
```

Comment:
- [ ] [OPTIONAL] For a good practice it is always better to remove the printings after testing or debugging your app